pkg/sql/schemachanger/scbuild/internal/scbuildstmt/process.go

```diff
@@ -122,45 +35,55 @@ func IsFullySupportedWithFalsePositive(
 	n tree.Statement,
 	activeVersion clusterversion.ClusterVersion,
 	mode sessiondatapb.NewSchemaChangerMode,
-) (ret bool) {
-	return isFullySupportedWithFalsePositiveInternal(supportedStatements, reflect.TypeOf(n),
-		reflect.ValueOf(n), mode, activeVersion)
-}
-
-// isFullySupportedWithFalsePositiveInternal determines whether a stmt is
-// fully supported, with false positive, in the declarative schema changer,
-// given mode the active cluster version.
-func isFullySupportedWithFalsePositiveInternal(
-	stmtSupportMap map[reflect.Type]supportedStatement,
-	stmtType reflect.Type,
-	stmtValue reflect.Value,
-	mode sessiondatapb.NewSchemaChangerMode,
-	activeVersion clusterversion.ClusterVersion,
 ) bool {
 	if mode == sessiondatapb.UseNewSchemaChangerOff {
 		return false
 	}
 
-	info, ok := stmtSupportMap[stmtType]
-	if !ok {
+	switch typedN := n.(type) {
+	case *tree.AlterTable:
+		return alterTableChecks(typedN, activeVersion)
+	case *tree.CreateIndex:
+		return activeVersion.IsActive(clusterversion.V23_1)
+	case *tree.DropDatabase:
```

Comment:
nit: I've found that it's much more readable to combine cases that are the same into a single case. Once the line starts to get very long, go fmt will wrap it automatically.

```go
switch {
case *tree.DropOwnedBy, *tree.DropSchema, *tree.DropSequence, *tree.DropType:
     return true
}


```